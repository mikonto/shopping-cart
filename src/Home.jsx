import styled from "styled-components";

const StyledHome = styled.div`
  background: white;
  flex-grow: 1;
  padding: 4px;
`;

const Home = () => {
  return (
    <>
      <StyledHome>Home</StyledHome>
    </>
  );
};

export default Home;
