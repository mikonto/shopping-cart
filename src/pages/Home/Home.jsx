import styled from "styled-components";

const StyledHome = styled.div`
  flex-grow: 1;
  padding: 12px;
`;

const Title = styled.h1`
  background: #f6f6f6;
  padding: 12px;
  margin: 8px;
`;

const Home = () => {
  return (
    <>
      <Title>Home</Title>
      <StyledHome></StyledHome>
    </>
  );
};

export default Home;
