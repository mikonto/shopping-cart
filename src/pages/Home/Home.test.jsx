import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Home from "./Home";

describe("Home component", () => {
  it("renders the correct title", () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const title = screen.getByRole("heading", { name: /welcome to webshop!/i });
    expect(title).toBeInTheDocument();
  });

  it("renders the correct paragraph text", () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    const paragraphText =
      /Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Cras pulvinar mattis nunc sed./s;
    const paragraph = screen.getByText(paragraphText);
    expect(paragraph).toBeInTheDocument();
  });

  it("renders the button and navigates correctly", () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const button = screen.getByRole("button", { name: "Shop now" });
    expect(button).toBeInTheDocument();

    const link = screen.getByRole("link", { name: "Shop now" });
    expect(link).toHaveAttribute("href", "/products");
  });

  it("it renders correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
