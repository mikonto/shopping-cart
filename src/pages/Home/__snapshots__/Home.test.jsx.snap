// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > it renders correctly 1`] = `
<div>
  <div
    class="sc-bcXGCL bOHEED"
  >
    <div
      class="sc-gswONk gPsxiH"
    >
      <h1
        class="sc-dkrGBB dZvstf"
      >
        Welcome to Webshop!
      </h1>
    </div>
    <div
      class="sc-gswONk gPsxiH"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Cras pulvinar mattis nunc sed.
      </p>
    </div>
    <div
      class="sc-gswONk gPsxiH"
    >
      <a
        href="/products"
      >
        <button
          class="sc-hLBatu fZVKwj"
        >
          Shop now
        </button>
      </a>
    </div>
  </div>
</div>
`;
